### PR TITLE
✨ JS-609: Event Delegation

### PR DESCRIPTION
**Pull request by ct-aa**
# Related Tickets
- [JS-609](https://codeandtheory.atlassian.net/browse/JS-609)

# Description
Added the **Event Delegation** challenge solutions.

# Validation
* Open the **Event Delegation** folder.
* Open the app.js.
* The event listener should only apply to the input elements.

# Risks
N/A

### CODE REVIEW

File: Event Delegation/app.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+let section = document.getElementsByTagName('section')[0];
```
Comment (lycoangelo) [inline Event Delegation/app.js]: We can replace this with a `querySelector` so we don't need to access the `section` via index.

File: Event Delegation/index.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>JavaScript and the DOM</title>
+  </head>
+  <link rel="stylesheet" href="style.css" />
```
Comment (lycoangelo) [inline Event Delegation/index.html]: `style.css` doesn't exist.

**Review by lycoangelo — CHANGES_REQUESTED** [review]

File: Event Delegation/app.js (hunk 1 above)
Comment (ct-aa) [reply Event Delegation/app.js]: Thanks, @lycoangelo, This is now updated.

File: Event Delegation/index.html (hunk 2 above)
Comment (ct-aa) [reply Event Delegation/index.html]: Thanks, @lycoangelo, This is now updated.

File: Event Delegation/app.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+let inputItems = document.querySelectorAll('input');
+
```
Comment (lycoangelo) [inline Event Delegation/app.js:2]: Please use `const` since the variable value was not reassigned.

File: Event Delegation/app.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+for (let i = 0; i < inputItems.length; i++) {
+	inputItems[i].addEventListener('click', (e) => {
+		if (e.target.tagName === 'INPUT') {
+			e.target.style.backgroundColor = 'rgb(255, 255, 0)';
+		}
+	});
+}
```
Comment (lycoangelo) [inline Event Delegation/app.js:7]: Please format codes properly.

File: Event Delegation/app.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+let inputItems = document.querySelectorAll('input');
+
+for (let i = 0; i < inputItems.length; i++) {
+	inputItems[i].addEventListener('click', (e) => {
+		if (e.target.tagName === 'INPUT') {
+			e.target.style.backgroundColor = 'rgb(255, 255, 0)';
+		}
```
Comment (lycoangelo) [inline Event Delegation/app.js]: Will there be an issue if we remove the `if` statement? And just run `e.target.style.backgroundColor = 'rgb(255, 255, 0)';` right away?

**Review by lycoangelo — CHANGES_REQUESTED** [review]

File: Event Delegation/app.js (hunk 3 above)
Comment (ct-aa) [reply Event Delegation/app.js:2]: Thanks, @lycoangelo, This is now updated.

File: Event Delegation/app.js (hunk 4 above)
Comment (ct-aa) [reply Event Delegation/app.js:7]: Thanks, @lycoangelo, This is now updated.

File: Event Delegation/app.js (hunk 5 above)
Comment (ct-aa) [reply Event Delegation/app.js]: Thanks, @lycoangelo, Removing the if statement has no issue. It is now removed.

File: Event Delegation/app.js (hunk 4 above)
Comment (lycoangelo) [reply Event Delegation/app.js:7]: @ct-aa, ideally, we should use 2 spaces (not tabs) for JS code.

Comment (ct-aa) [reply Event Delegation/app.js:7]: Thanks, @lycoangelo, This is now updated.

**Review by lycoangelo — APPROVED** [review]

**Pull request merged**